packages/algob/src/internal/deployer.ts

```diff
@@ -586,6 +593,31 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
     return lsigInfo;
   }
 
+  /**
+   * Stores logic signature info in checkpoint for contract mode
+   * @param name ASC name
+   * @param scTmplParams: Smart contract template parameters (used only when compiling PyTEAL to TEAL)
+   */
+  async addLsigCheckpoint (name: string, scTmplParams?: SCParams): Promise<LsigInfo> {
+    this.assertNoAsset(name);
+    let lsigInfo = {} as any;
+    try {
+      const lsig = await getLsig(name, this.algoOp.algodClient, scTmplParams);
+      lsigInfo = {
+        creator: lsig.address(),
+        contractAddress: lsig.address(),
+        lsig: lsig
+      };
+    } catch (error) {
+      this.persistCP();
+
+      console.log(error);
+      throw error;
+    }
+    this.cpData.registerLsig(this.networkName, name, lsigInfo);
+    return lsigInfo;
+  }
```

Comment:
I think most of  the code here is same with `mkDelegatedLsig` (except `lsig.sign`). Can we move code to a common function (maybe `_mkLsig`) & reuse in both?

Reply:
how do we remove `lsig.sign` in a common function ?

Reply:
I was thinking `_mkLsig(lsig, signer?)`. If signer is passed we store delegated lsig. Could something like this work? Could you please check?

Reply:
ok